src/pages/settings/index.jsx

```diff
@@ -22,15 +22,18 @@ import DevModeContext from "../../contexts/devMode";
 import ColorModeContext from "../../contexts/colorMode";
 import { localSettings, localSettingsKeys } from "../../localeStorageManager";
 import FooterWithLinks from "../../components/Footer";
+import countryNames  from "../../assets/countries.json";
+import { capitaliseName } from "../../utils";
 
 export default function Settings() {
   const { t, i18n } = useTranslation();
   const theme = useTheme();
   const colorMode = React.useContext(ColorModeContext);
-
+  const localData= localSettings.fetch();
   const [language, setLanguage] = React.useState(i18n.language);
   const { devMode, setDevMode, visiblePages, setVisiblePages } =
     React.useContext(DevModeContext);
+  const [selectedCountry, setSelectedCountry]= React.useState(localData['country'])
 
```

Comment:
```suggestion
  const [language, setLanguage] = React.useState(i18n.language);
  const { devMode, setDevMode, visiblePages, setVisiblePages } =
    React.useContext(DevModeContext);
  const [selectedCountry, setSelectedCountry]= React.useState(() => localSettings.fetch()['country'])

```

To avoid fetching localeStorage every render whereas we only need it at initialization